samples/template-operator/operator/controllers/sample_controller.go

```diff
@@ -62,13 +61,18 @@ func (r *SampleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+// initReconciler injects the required configuration into the declarative reconciler.
 func (r *SampleReconciler) initReconciler(mgr ctrl.Manager) error {
+	manifestResolver := &ManifestResolver{}
 	return r.Inject(mgr, &v1alpha1.Sample{},
+		declarative.WithManifestResolver(manifestResolver),
 		declarative.WithResourceLabels(map[string]string{"sampleKey": "sampleValue"}),
 		declarative.WithObjectTransform(transform),
+		declarative.WithResourcesReady(true),
 	)
 }
 
+// transform modifies the resources based on some criteria, before installation.
```

Comment:
I think we should mention that this is just a sample and people can basically to anything here